x-pack/legacy/plugins/maps/public/angular/map_controller.js

```diff
@@ -382,7 +384,7 @@ app.controller(
       }
 
       const nextIndexPatternIds = getQueryableUniqueIndexPatternIds(store.getState());
-      if (nextIndexPatternIds !== prevIndexPatternIds) {
+      if (!_.isEqual(nextIndexPatternIds, prevIndexPatternIds)) {
```

Comment:
Can we keep this as `!==` check? This method gets called on each store change so we want it to be as light weight as possible. We are using re-select so the array reference will be the same unless underlying store changes need to create a new array

Reply:
Since `getQueryableUniqueIndexPatternIds` always creates a new array, isn't this always true?

Reply:
`getQueryableUniqueIndexPatternIds` uses [reselect](https://github.com/reduxjs/reselect) so it only creates a new array when the parameters change

https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/maps/public/selectors/map_selectors.js#L191
```
export const getQueryableUniqueIndexPatternIds = createSelector(getLayerList, layerList => {
  const indexPatternIds = [];
  layerList.forEach(layer => {
    indexPatternIds.push(...layer.getQueryableIndexPatternIds());
  });
  return _.uniq(indexPatternIds);
});
```